packages/Upgrade/src/api/repositories/ErrorRepository.ts

```diff
@@ -18,15 +19,30 @@ export class ErrorRepository extends Repository<ExperimentError> {
     return result.raw && result.raw[0];
   }
 
-  public async paginatedFind(limit: number, offset: number): Promise<ExperimentError[]> {
-    return this.createQueryBuilder('error')
+  public async paginatedFind(limit: number, offset: number, filter: SERVER_ERROR): Promise<ExperimentError[]> {
+    let queryBuilder = this.createQueryBuilder('error')
       .skip(offset)
       .take(limit)
-      .orderBy('error.createdAt', 'DESC')
+      .orderBy('error.createdAt', 'DESC');
+    if (filter) {
+      queryBuilder = queryBuilder
+        .where('error.type = :filter', { filter });
+    }
+    return queryBuilder
       .getMany()
       .catch((error: any) => {
         const errorMsg = repositoryError('ErrorRepository', 'paginatedFind', { limit, offset }, error);
         throw new Error(errorMsg);
       });
   }
+
+  public getTotalLogs(filter: SERVER_ERROR): Promise<number> {
+    return this.createQueryBuilder('error')
+      .where('error.type = :filter', { filter })
```

Comment:
Do we need to put this in if condition in case of no filters?

Reply:
No because if we won't provide any filter then it will be return through service directly

Reply:
ok